app/browser/reducers/tabsReducer.js

```diff
@@ -126,7 +71,23 @@ const tabsReducer = (state, action, immutableAction) => {
     case appConstants.APP_TAB_CREATED:
       state = tabState.maybeCreateTab(state, action)
       break
-    case appConstants.APP_TAB_MOVED: {
+    case appConstants.APP_TAB_ATTACHED:
+      state = tabs.updateTabsStateForAttachedTab(state, action.get('tabId'))
+      break
+    case appConstants.APP_TAB_WILL_ATTACH: {
+      const tabId = action.get('tabId')
+      const tabValue = tabState.getByTabId(state, tabId)
+      if (!tabValue) {
+        break
+      }
+      const oldWindowId = tabState.getWindowId(state, tabId)
+      state = tabs.updateTabsStateForWindow(state, oldWindowId)
+      break
+    }
+    case appConstants.APP_TAB_MOVED:
+      state = tabs.updateTabsStateForAttachedTab(state, action.get('tabId'))
```

Comment:
`appConstants.APP_TAB_MOVED` can be grouped with `appConstants.APP_TAB_ATTACHED` (since they have the same code)

Reply:
they are 2 diff things though so I don't think they should be.